libvast/vast/taxonomies.hpp

```diff
@@ -28,8 +28,14 @@ namespace vast {
 /// Maps concept names to the fields or concepts that implement them.
 using concepts_type = std::unordered_map<std::string, std::vector<std::string>>;
 
+/// Converts a data object and fills it in out.
+caf::error convert(const data& d, concepts_type& out);
+
+/// Extracts a concept definition from a data object.
+caf::error extract_concepts(const data& d, concepts_type& out);
+
 /// Extracts a concept definition from a data object.
-caf::expected<concepts_type> to_concepts(const data& d);
+caf::expected<concepts_type> extract_concepts(const data& d);
```

Comment:
Do we need both overloads? I could image that the latter suffices.

Reply:
The former is required to accumulate concepts from multiple sources. In principle we could also create a separate `merge` function for `concepts_type`, but why shouldn't the function already behave the way we need it?